### PR TITLE
[pvr] bye, bye libXBMC_pvr.cpp

### PR DESCRIPTION
**Pull request by AlwinEsch**
Hi all,

my big add-on rework takes to much time and that's why I bring this smaller one. This remove the usage of shared pvr library on add-on's and the communication is done direct on header with the function table from Kodi (What was before on the cpp of lib). 

If this is OK and works can also this way be done for the other libraries.

### CODE REVIEW

**Comment by AlwinEsch** [comment]
jenkins build this please

**Comment by FernetMenta** [comment]
great! what are your plans for integrations? You don't plan this for v17, do you?

**Comment by AlwinEsch** [comment]
If acceptable for you and the others we can do it for v17, was not sure if OK on the Beta.

Normally are the old who use the library (if lib present) are also working, nothing changed on the function table itself.

**Comment by AlwinEsch** [comment]
If all are swapped to the new way can be the `..._RegisterMe` and `..._UnRegisterMe` removed, but then all add-ons need a Version change to prevent compatibility faults.

**Comment by MartijnKaijser** [comment]
Then that's a no.

**Comment by ksooo** [comment]
@AlwinEsch sorry, API changes for v17 are a no go as we are already in beta.

**Comment by AlwinEsch** [comment]
@ksooo no problem thought the same, and sorry that it come so late.

**Comment by ksooo** [comment]
@AlwinEsch no problem, you are doing a great job. Keep it up.

**Comment by AlwinEsch** [comment]
OK to have on end all library parts in one request or better to have every part in a own pull request?

**Comment by FernetMenta** [comment]
I think separate PRs are easier to review

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/libXBMC_pvr.h [hunk 1]
```diff
@@ -27,29 +27,78 @@
...
-
 #define DVD_TIME_BASE 1000000
 
 //! @todo original definition is in DVDClock.h
 #define DVD_NOPTS_VALUE 0xFFF0000000000000
 
+namespace V1
+{
+namespace KodiAPI
+{
```
Comment (Paxxi) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/libXBMC_pvr.h]: I think KodiAPI should be the outer namespace as it won't change and have the version as the second level

**Comment by Paxxi** [comment]
Nice work, seems like a sensible way to handle it

**Comment by AlwinEsch** [comment]
About the:
`
If all are swapped to the new way can be the ..._RegisterMe and ..._UnRegisterMe removed, but then all add-ons need a Version change to prevent compatibility faults.
`
above. Is not needed.

All add-ons need only recompile and nothing else :)

Acceptable to bring this and the changes for other libraries in?

**Comment by AlwinEsch** [comment]
jenkins build addons please

**Comment by AlwinEsch** [comment]
jenkins build this please

**Comment by fritsch** [comment]
jenkins build this please

**Comment by AlwinEsch** [comment]
jenkins build this please

**Comment by AlwinEsch** [comment]
OK to bring in?

jenkins build this please

**Comment by AlwinEsch** [comment]
jenkins build this with addons please

**Pull request closed without merging**